app/database/operator/server_data_operator/transformers/post.ts

```diff
@@ -151,3 +153,36 @@ export const transformPostsInChannelRecord = ({action, database, value}: Transfo
         fieldsMapper,
     }) as Promise<PostsInChannelModel>;
 };
+
+/**
+ * transformPostRecords: Prepares records of the SERVER database 'ScheduledPosts' table for update or create actions.
+ */
+export const transformSchedulePostsRecord = ({action, database, value}: TransformerArgs): Promise<ScheduledPostModel> => {
+    const emptyFileInfo: FileInfo[] = [];
+    const emptyPostMetadata: PostMetadata = {};
+    const raw = value.raw as ScheduledPost;
+
+    const fieldsMapper = (scheduledPost: ScheduledPostModel) => {
+        scheduledPost._raw.id = scheduledPost.id;
+        scheduledPost.rootId = raw?.root_id ?? '';
+        scheduledPost.message = raw?.message ?? '';
+        scheduledPost.channelId = raw?.channel_id ?? '';
+        scheduledPost.files = raw?.files ?? emptyFileInfo;
+        scheduledPost.metadata = raw?.metadata ?? emptyPostMetadata;
+        if (raw.priority) {
+            scheduledPost.metadata.priority = raw.priority;
+        }
+        scheduledPost.updateAt = raw.update_at ?? Date.now();
+        scheduledPost.scheduledAt = raw.scheduled_at ?? Date.now();
```

Comment:
We want the scheduled at to default to now?

Reply:
Nice catch, will remove it.